copylot/hardware/cameras/abstract_camera.py

```diff
@@ -1,5 +1,114 @@
-import abc
+from abc import ABCMeta, abstractmethod
 
 
-class AbstractCamera(metaclass=abc.ABCMeta):
-    pass
+class AbstractCamera(metaclass=ABCMeta):
+
```

Comment:
Do we need a `device_ID()`? Sometimes it's useful referencing the cameras based on the serial number, but hard to say if it should be part of the abstract camera class. I know serial number would be something that doesn't change regardless of connection type (i.e USB, COM port, etc). @AhmetCanSolak @YangLiujames

Reply:
Which camera providers expose serial number of cameras through their APIs?

Reply:
basler, flir, teledyne photometrix, thorlabs. In my experience most cameras have that setting since it is easier to initialize a camera through the serial number than USB specially when you have multiple cameras connected.

Reply:
how about having a class member called `serial_number`? which  can be assigned in different ways in different child classes. What do you think @edyoshikun ?

Reply:
I added the device_id property @edyoshikun , will merge now but if you have any change request let me know. I can always do follow-up PRs.